### ISSUE TITLE
`@stylistic/selector-descendant-combinator-no-non-space` never reports an illegal combinator such as `( )`

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#125]
Found by unskipping the second of the two skipped cases in `lib/rules/selector-descendant-combinator-no-non-space/index.test.js`, neither of which has an issue behind it.

### What minimal example or steps are needed to reproduce the bug?

```css
.foo (  ) .bar {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/selector-descendant-combinator-no-non-space": true
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json
```

### Which Stylelint-related dependencies are you using?

```json
{
  "@stylistic/stylelint-plugin": "5.3.0",
  "stylelint": "17.14.1"
}
```

### What did you expect to happen?

`Unexpected " (  )"` at line 1, column 5. The rule is there to say that a descendant combinator is a single space and nothing else, and `(  )` is neither a space nor a combinator CSS knows.

### What actually happened?

Nothing is reported.

### Do you have a proposal to fix the bug?

`postcss-selector-parser` gives such a combinator a `value` of `" (  )"` — leading whitespace and all — and the rule's first line drops it:

```js
if (combinatorNode.value !== ` `) return
```

A combinator whose `value` holds whitespace and yet is not a single space is exactly the illegal one. Every combinator CSS defines carries no whitespace in its `value` at all: `>`, `+`, `~` and `||` keep their spaces in `spaces.before` and `spaces.after`, and so do `>>>` and `/deep/`. A descendant combinator written with more than one space is `" "` with the rest in `spaces.before`, which is what the rule already reports.

So the guard can let through anything whose `value` is a single space or contains whitespace, and the existing `/^\s+$/u` guard on the fix already declines to touch what it cannot repair — the case is unfixable, and a warning is all it should get.

`@stylistic/selector-descendant-combinator-no-non-space` also ignores selectors carrying comments, tracked separately.

**Comment by firefoxic** [comment]
Half of the proposal above is wrong, and the fix departs from it.

> the existing `/^\s+$/u` guard on the fix already declines to touch what it cannot repair

It does not. A fixer cannot decline from the inside: Stylelint's `isFixApplied()` calls it, discards whatever it returns and counts the problem as fixed either way. So widening the guard on its own would have handed every illegal combinator a fixer that silences its own warning and leaves the selector as it was — a fresh instance of #126, which is what that issue turned out to be about. Measured: the case passes without the widened guard for want of a warning, and fails with the guard alone because the warning does not survive its own fix.

The decision therefore moves out of the fixer, and `report` is handed no fixer at all where the combinator cannot be repaired.

One thing the report understated. The second condition — the run of surplus whitespace — belongs to the descendant combinator and says nothing about an illegal one, which is a problem whatever it is made of. Reading the two apart brings `.foo ( ) .bar` and `.foo () .bar` in alongside `.foo (  ) .bar`; only the last of the three would have been reported otherwise, though the first is the one this issue is named after.

**Comment by firefoxic** [comment]
Two corrections to the comment above.

It says the case "passes without the widened guard". It does not — it fails there too, for want of a warning at all (`reject: expected 1 warning, got 0`). Both states fail, which is what the commit says; the word in that sentence was simply wrong.

And widening the guard turned out to expose a hole elsewhere. A Less CSS guard reaches the parser as the same shape — `.a:hover when (1 = 1)` yields a combinator whose `value` is `" (1 = 1)"` — and `isStandardSyntaxSelector` catches a guard only by accident: one of its rules because the selector ends in a parenthesis and carries no colon, another because the condition names a variable. A guard answering to neither was being read as ordinary CSS by every rule built on that util. Harmless while a combinator with such a `value` went unread, but from here it would have been a warning on valid Less that nothing could clear. The fix teaches the util the guard itself, and the two Less cases the rule already had — both naming a variable, so both caught by accident — are joined by ones that name none.
